added DID format (#2)
richanna/secevent: draft-ietf-secevent-subject-identifiers.md

    target: http://www.iana.org/assignments/jwt
    author:
      org: IANA
  DID:
    title: Decentralized Identifiers (DIDs) v1.0

>     target: https://www.w3.org/TR/did-core/

Reviewer: This may not be the correct durable link, because W3C might publish a later version of this spec. I suggest to consult with the W3C liaison, @wseltzer.

Author: I believe this is the recommended stable URL for the document, but I am happy to change it if there's a better one.

Reviewer 2: thanks, /TR links are stable

~~~
{: #figexamplesubidopaque title="Example: Subject Identifier in the Opaque Identifier Format"}

> ### Decentralized Identifier (DID) Format {#sub-id-did}
> The Decentralized Identifier Format identifies a subject using a Decentralized Identifier (DID) URL, identified with a DID URL as defined in {{DID}}.  Subject Identifiers in this format MUST contain a `uri` member whose value is the DID URL for the subject. The value of the `uri` member MAY be any valid DID URL including a bare DID. The `uri` member is REQUIRED and MUST NOT be null or empty. The Decentralized Identifier Format is identified by the name `did`.

Reviewer 3: 1. Replaced `uri` with `url`, since these are DID UR *L* s. Could be convinced to use `uri` though, if there's a good reason. (e.g., existing convention within the DID community)
2. The "MAY" was confusing to me; tweaked the language to be a bit clearer (IMHO; happy to iterate on it if needed)

```suggestion
### Decentralized Identifier (DID) Format {#sub-id-did}
The Decentralized Identifier Format identifies a subject using a Decentralized Identifier (DID) URL as defined in {{DID}}.  Subject Identifiers in this format MUST contain a `url` member whose value is the DID URL for the subject. The value of the `url` member MUST be a valid DID URL as defined in {{DID}}, and MAY be a bare DID. The `url` member is REQUIRED and MUST NOT be null or empty. The Decentralized Identifier Format is identified by the name `did`.
```

Reviewer 3: Should this be restricted to *absolute* DID URLs? We should specify, either way. Maybe "absolute only" is implied, but it seems worth clarifying since did-core talks about relative DID URLs.

Reviewer 3: > the DID URL for the subject

Should this be "*a* DID URL for the subject"? Can a subject have more than one DID URL, or do different DID URLs necessarily refer to different subjects?

> 
> Below is a non-normative example Subject Identifier for the Decentralized Identifier Format:
> 
> ~~~
> {
>   "format": "did",
>   "uri": "did:example:123456"
> }
> ~~~
> {: #figexamplesubiddid title="Example: Subject Identifier for the Decentralized Identifier Format"}
> 

Reviewer 3: Changes `uri` to `url` in accordance with my above suggestion, and adds an example that uses a more complex DID URL.

```suggestion

Below are non-normative example Subject Identifiers for the Decentralized Identifier Format:

~~~
{
  "format": "did",
  "url": "did:example:123456"
}
~~~
{: #figexamplesubiddidbare title="Example: Subject Identifier for the Decentralized Identifier Format, identifying a subject with a bare DID"}

~~~
{
  "format": "did",
  "url": "did:example:123456/did/url/path?versionId=1"
}
~~~
{: #figexamplesubiddidcomplex title="Example: Subject Identifier for the Decentralized Identifier Format, identifying a subject with a DID URL with non-empty path and query components"}
```

~~~
{: #figexamplesubidopaque title="Example: Subject Identifier in the Opaque Identifier Format"}

### Decentralized Identifier (DID) Format {#sub-id-did}

> The Decentralized Identifier Format identifies a subject using a Decentralized Identifier (DID) URL as defined in {{DID}}.  Subject Identifiers in this format MUST contain a `url` member whose value is a DID URL for the subject. The value of the `url` member MUST be a valid DID URL and MAY be a bare DID. The `url` member is REQUIRED and MUST NOT be null or empty. The Decentralized Identifier Format is identified by the name `did`.

Author: ```suggestion
The Decentralized Identifier Format identifies a subject using a Decentralized Identifier (DID) URL as defined in {{DID}}.  Subject Identifiers in this format MUST contain a `url` member whose value is a DID URL for the DID Subject being identified. The value of the `url` member MUST be a valid DID URL and MAY be a bare DID. The `url` member is REQUIRED and MUST NOT be null or empty. The Decentralized Identifier Format is identified by the name `did`.
```

* Change Controller: IETF
* Defining Document(s): {{sub-ids}} of this document.

#### Decentralized Identifier Format

> * Type Name: `did`

Reviewer 3: Should this be `did` or `did_url`? If there is a reasonable chance that we will want to define a subject identifier format that specifically allows only bare DIDs, then I think `did_url` would be better. But my sense is that a profiling spec/implementer could use this format and define constraints on the types of DID URLs that are allowed/accepted. (or those constraints may be implied by the context) If so, then leaving this as `did` makes sense. But I'd appreciate confirmation from someone more familiar with the DID world.

Author: It should be `did`. Profiles can constrain if they want already and a new format could be defined for bare DIDs if necessary. All DID URLs are usable as DIDs, and the DID WG decided that the top level component should be the more flexible format.

Reviewer 3: > All DID URLs are usable as DIDs

Interesting. That raises another question: is the subject identified by a `did` subject identifier the DID Subject identified by the bare DID portion of the DID URL in the `url` property, or is it the entity/resource that the DID URL resolves to? For example, suppose Alice has a DID URL that resolves to a specific verification method described within a DID Document for which Alice is the DID Subject. If Alice used this DID URL in a `did` subject identifier, should that be interpreted as identifying Alice, or the verification method? Or does it depend on context? Either way, a non-normative clarification would be helpful, I think. I can write that, just need to know which way to write it.

* Type Description: Subject identifier based on a Decentralized Identifier (DID) URL.

> * Change Controller: IETF

Reviewer 3: Hmm. Is this right? Should this be W3C instead of IETF? Is this enough of a reason to pull this out of this doc and register it in a doc published by W3C?

Author: As far as I know this is correct, but we can bring it up in IESG review if needed. @wseltzer do you have input here, for who's the change controller for something registered in an IETF document that refers to a W3C document?
